### PR TITLE
Auto generate typed graphql hooks

### PR DESCRIPTION
**Pull request by timothybest-usds**
## Related Issue or Background Info

- Strictly typing our API can better enforce backwards compatibility of our application and could have in part prevented some of our previous production issues

## Changes Proposed

- remove duplicate graphql operations
- Add codegen command to generate typed hooks
- Add CI check to ensure the generated code is kept up to date

## Additional Information

- https://www.graphql-code-generator.com
- Going to open separate PRs to start consuming this file. There appears to be a bit of work to do to consume this as is. For example or schema definition incorrectly defines most object IDs as nullable

### UI
- [ ] Any changes to the UI/UX are approved by design 
- [ ] Any new or updated content (e.g. error messages) are approved by design 

### Testing
- [ ] Includes a summary of what a code reviewer should verify

### Changes are Backwards Compatible
- [ ] Database changes are submitted as a separate PR
  - [ ] Any new tables that do not contain PII are accompanied by a GRANT SELECT to the no-PHI user
  - [ ] Any changes to tables that have custom no-PHI views are accompanied by changes to those views
        (including re-granting permission to the no-PHI user if need be)
  - [ ] Liquibase rollback has been tested locally using `./gradlew liquibaseRollbackSQL` or `liquibaseRollback`
- [ ] GraphQL schema changes are backward compatible with older version of the front-end

### Security
- [ ] Changes with security implications have been approved by a security engineer (changes to  authentication, encryption, handling of PII, etc.)
- [ ] Any dependencies introduced have been vetted and discussed

## Cloud
- [ ] DevOps team has been notified if PR requires ops support
- [ ] If there are changes that cannot be tested locally, this has been deployed to our Azure `test`, `dev`, or `pentest` environment for verification

### CODE REVIEW

File: frontend/package.json [hunk 1]
```diff
@@ -67,6 +67,7 @@
     "prebuild": "yarn compile-scss",
     "build": "INLINE_RUNTIME_CHUNK=false REACT_APP_CURRENT_COMMIT=$(git rev-parse HEAD) craco build",
     "postbuild": "echo 'pushstate: enabled' > build/Staticfile && echo $(git rev-parse HEAD) > build/static/commit.txt",
+    "codegen": "graphql-codegen --config codegen.yml",
```
Comment (zdeveloper) [inline frontend/package.json:70]: is this something we have to run manually everytime we change the graphql schema?

File: frontend/src/app/admin/Organization/AddOrganizationAdminFormContainer.tsx [hunk 2]
```diff
@@ -13,15 +13,7 @@ import { LoadingCard } from "../../commonComponents/LoadingCard/LoadingCard";
...
-
-const GET_ORGANIZATIONS_QUERY = gql`
-  query GetOrganizations($identityVerified: Boolean) {
-    organizations(identityVerified: $identityVerified) {
-      externalId
-      name
-    }
-  }
-`;
+import { GET_ORGANIZATIONS_QUERY } from "./TenantDataAccessFormContainer";
```
Comment (zdeveloper) [inline frontend/src/app/admin/Organization/AddOrganizationAdminFormContainer.tsx:16]: Nice job removing dupes

File: frontend/src/generated/graphql.tsx [hunk 3]
```diff
@@ -0,0 +1,5176 @@
+import { gql } from "@apollo/client";
```
Comment (zdeveloper) [inline frontend/src/generated/graphql.tsx:1]: this is sweet! cant wait to see how its gonna be consumed

**Review by zdeveloper — APPROVED** [review]
LGTM

File: frontend/package.json (hunk 1 above)
Comment (timothybest-usds) [reply frontend/package.json:70]: This is a really good flag. Thank you! It was going to be manual. Added a commit after seeing this to watch for changes and automatically update the codegen https://github.com/CDCgov/prime-simplereport/pull/2556/commits/bbbc4c366cca311b77490944055c8b66e50557c9

**Review by nickclyde — APPROVED** [review]
This is awesome, can't wait to clean up our components by consuming the generated queries and mutations!

**Pull request merged**